src/Functions/array/arrayElement.cpp

```diff
@@ -588,7 +588,7 @@ ColumnPtr FunctionArrayElement::executeGenericConst(
             col_nested, col_array->getOffsets(), safeGet<UInt64>(index) - 1, *col_res, builder);
     else if (index.getType() == Field::Types::Int64)
         ArrayElementGenericImpl::vectorConst<true>(
-            col_nested, col_array->getOffsets(), -safeGet<Int64>(index) - 1, *col_res, builder);
+            col_nested, col_array->getOffsets(), -(safeGet<Int64>(index) + 1), *col_res, builder);
```

Comment:
It still has signed integer overflow when index is max positive Int64.

Reply:
If I understand correctly, this branch is executed only when `index` is negative (we call `ArrayElementGenericImpl::vectorConst(...)` with `negative = true`), so there is no signed integer overflow.